### PR TITLE
[Feature]: Improve get catalog database table interface

### PR DESCRIPTION
**Pull request by hunter-cloud09**
Improve get catalog database table interface

re #83

### CODE REVIEW

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/DatabaseService.java [hunk 1]
```diff
@@ -18,28 +18,53 @@
...
      */
-    List<DatabaseInfo> selectByCatalogId(Integer catalogId);
+    R<Void> createDatabase(DatabaseDTO databaseDTO);
+
+    /**
+     * Get all database information.
+     *
+     * @return The list of all databases.
+     */
+    R<List<DatabaseVO>> getAllDatabases();
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/DatabaseService.java:53]:
```suggestion
    R<List<DatabaseVO>> listDatabases();
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/DatabaseService.java [hunk 2]
```diff
@@ -18,28 +18,53 @@
...
+
+    /**
+     * Get database information by catalog id.
+     *
+     * @return The list of databases.
+     */
+    R<List<DatabaseVO>> getDatabaseById(Integer catalogId);
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/DatabaseService.java:60]:
```suggestion
    R<List<DatabaseVO>> getDatabasesById(Integer catalogId);
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/DatabaseService.java [hunk 3]
```diff
@@ -18,28 +18,53 @@
...
+
+    /**
+     * Get all database information.
+     *
+     * @return The list of all databases.
+     */
+    R<List<DatabaseVO>> getAllDatabases();
+
+    /**
+     * Get database information by catalog id.
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/DatabaseService.java:56]:
```suggestion
     * Get databases given catalog id.
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/DatabaseService.java [hunk 4]
```diff
@@ -18,28 +18,53 @@
...
+     *
+     * @return The list of databases.
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/DatabaseService.java:58]:
```suggestion
     * @return The list of databases of given catalog.
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/DatabaseService.java [hunk 5]
```diff
@@ -18,28 +18,53 @@
...
+     */
+    R<List<DatabaseVO>> getDatabaseById(Integer catalogId);
+
+    /**
+     * Remove a database by its name.
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/DatabaseService.java:63]:
```suggestion
     * Drops database given database name.
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/DatabaseService.java [hunk 6]
```diff
@@ -18,28 +18,53 @@
...
+     *
+     * @param databaseDTO The drop database DTO.
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/DatabaseService.java:65]:
```suggestion
     * @param databaseDTO The dropping database.
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/DatabaseService.java [hunk 7]
```diff
@@ -18,28 +18,53 @@
...
+     * @return A response indicating the success or failure of the removal operation.
+     * @throws RuntimeException if the database is not found, or it is not empty.
+     */
+    R<Void> dropDatabase(DatabaseDTO databaseDTO);
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/DatabaseService.java:69]: Why does not the interface throw RuntimeException?

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/TableService.java [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+import org.apache.paimon.web.server.data.result.enums.Status;
+import org.apache.paimon.web.server.data.vo.TableVO;
+
+import java.util.List;
+
+/** Table service. */
+public interface TableService {
+
+    /**
+     * Creates a table in the database based on the provided TableInfo.
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/TableService.java:33]:
```suggestion
     * Creates a table in the database given ${@link TableInfo}.
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/TableService.java [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+     * @param databaseName The name of the database.
+     * @param tableName The name of the table.
+     * @param columnName The name of the column to be dropped.
+     * @return The result indicating the success or failure of the operation.
+     */
+    R<Void> dropColumn(
+            String catalogName, String databaseName, String tableName, String columnName);
+
+    /**
+     * Modify a column in a table.
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/TableService.java:62]: Why not alter a table? This should not only modify column.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/TableService.java [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+     * @return Response object containing a list of {@link TableDTO} representing the tables.
+     */
+    R<List<TableDTO>> getAllTables();
+
+    /**
+     * Query tables by catalog id and database name.
+     *
+     * @return Response object containing a list of {@link TableVO} representing the tables.
+     */
+    R<List<TableVO>> getTablesByCondition(Integer catalogId, String databaseName);
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/TableService.java:142]: Any difference between this method and `queryTablesByCondition`?

**Review by SteNicholas — CHANGES_REQUESTED** [review]
@hunter-cloud09, thanks for contribution. I left some comments. PTAL.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/TableService.java (hunk 10 above)
Comment (hunter-cloud09) [reply paimon-web-server/src/main/java/org/apache/paimon/web/server/service/TableService.java:142]: queryTablesByCondition is used to search table by name fuzzy. getTablesByCondition is used to query all tables under a database under a catalog

**Comment by zhangmo8** [comment]
#92

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/DatabaseController.java [hunk 11]
```diff
@@ -84,28 +62,18 @@ public R<Void> createDatabase(@RequestBody DatabaseDTO databaseDTO) {
...
+        return databaseService.listDatabases();
+    }
+
+    /**
+     * Get databases given catalog id.
+     *
+     * @return The list of databases of given catalog id.
+     */
+    @GetMapping("/getDatabasesById/{catalogId}")
+    public R<List<DatabaseVO>> getDatabasesById(@PathVariable Integer catalogId) {
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/controller/DatabaseController.java:75]:
```suggestion
    public R<List<DatabaseVO>> getDatabasesByCatalogId(@PathVariable Integer catalogId) {
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/DatabaseService.java [hunk 12]
```diff
@@ -18,28 +18,53 @@
...
+     * @return The list of all databases.
+     */
+    R<List<DatabaseVO>> listDatabases();
+
+    /**
+     * Get databases given catalog id.
+     *
+     * @return The list of databases of given catalog.
+     */
+    R<List<DatabaseVO>> getDatabasesById(Integer catalogId);
```
Comment (SteNicholas) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/service/DatabaseService.java:61]:
```suggestion
    R<List<DatabaseVO>> getDatabasesByCatalogId(Integer catalogId);
```

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/service/TableService.java (hunk 10 above)
Comment (SteNicholas) [reply paimon-web-server/src/main/java/org/apache/paimon/web/server/service/TableService.java:142]: @hunter-cloud09, you could merge the two methods and use table object as parameter.

**Review by SteNicholas — CHANGES_REQUESTED** [review]
@hunter-cloud09, thanks for updates. I left some comments for this pull request. PTAL.

**Comment by hunter-cloud09** [comment]
@SteNicholas I have fixed issues. PTAL.

File: paimon-web-server/src/main/java/org/apache/paimon/web/server/data/vo/UserInfoVO.java [hunk 13]
```diff
@@ -7,7 +7,11 @@
  * "License"); you may not use this file except in compliance
  * with the License.  You may obtain a copy of the License at
  *
+<<<<<<< HEAD
```
Comment (zhangmo8) [inline paimon-web-server/src/main/java/org/apache/paimon/web/server/data/vo/UserInfoVO.java]: resolve conflicts

**Review by SteNicholas — CHANGES_REQUESTED** [review]
@hunter-cloud09, thanks for the updates. I left some comments for this update. PTAL.

**Review by SteNicholas — APPROVED** [review]
LGTM. Thanks for @hunter-cloud09 updates.

**Pull request merged**